### PR TITLE
pythonPackages.criticality-score: init at 1.0.7

### PR DESCRIPTION
**Pull request by wamserma**
###### Motivation for this change

Compute the [criticality score](https://github.com/ossf/criticality_score) of https://github.com/NixOS/nixpkgs.

Result:
```
criticality_score --repo github.com/NixOS/nixpkgs
name: nixpkgs
url: https://github.com/NixOS/nixpkgs
language: Nix
created_since: 217
updated_since: 0
contributor_count: 4790
org_count: 9
commit_frequency: 703.1
recent_releases_count: 6
updated_issues_count: 13120
closed_issues_count: 9727
comment_frequency: 2.0
dependents_count: 33204
criticality_score: 0.96204
```

Seems like nixpkgs is pretty important.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 108794"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wamserma** [comment]
This PR includes https://github.com/NixOS/nixpkgs/pull/108792 which should be merged first.

**Comment by SuperSandro2000** [comment]
> This PR includes #108792 which should be merged first.

Next time just include it in this.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1426,6 +1426,8 @@ in {
 
   credstash = callPackage ../development/python-modules/credstash { };
 
+  criticality_score = callPackage ../development/python-modules/criticality_score { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:1429]:
```suggestion
  criticality-score = callPackage ../development/python-modules/criticality-score { };
```

Python packages need to follow the pep that converts _ to -.

File: pkgs/development/python-modules/criticality_score/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0i811a27i87z3j1rw0dwrnw8v0ckbd918ms6shjawhs4cnb1c6x8";
+  };
+
+  propagatedBuildInputs = [ PyGithub python-gitlab ];
+
+  # tests are not included in pypi distributions and would require lots of
+  # optional dependencies
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/criticality_score/default.nix:16]:
```suggestion
  doCheck = true;
  pythonImportsCheck = [ "criticality_score" ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108794` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
17 packages built:
</summary>
<ul>
<li>cligh</li>
<li>mathlibtools (python38Packages.mathlibtools)</li>
<li>python37Packages.PyGithub</li>
<li>python37Packages.criticality_score</li>
<li>python37Packages.howdoi</li>
<li>python37Packages.keep</li>
<li>python37Packages.mathlibtools</li>
<li>python38Packages.PyGithub</li>
<li>python38Packages.criticality_score</li>
<li>python38Packages.howdoi</li>
<li>python38Packages.keep</li>
<li>python39Packages.PyGithub</li>
<li>python39Packages.criticality_score</li>
<li>python39Packages.howdoi</li>
<li>python39Packages.keep</li>
<li>python39Packages.mathlibtools</li>
<li>sourcehut.dispatchsrht</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.criticality_score:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python37Packages.keep:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.criticality_score:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.keep:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.criticality_score:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.keep:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>

File: pkgs/development/python-modules/criticality_score/default.nix (hunk 2 above)
Comment (wamserma) [reply pkgs/development/python-modules/criticality_score/default.nix]: Thanks for the review. Updated PR.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (wamserma) [reply pkgs/top-level/python-packages.nix]: Thanks for the review. Updated PR.

**Comment by wamserma** [comment]
> > This PR includes #108792 which should be merged first.
> 
> Next time just include it in this.

The bump was required to make this run, but also has some downstream packages so I split it off into an own PR.

**Comment by SuperSandro2000** [comment]
> The bump was required to make this run, but also has some downstream packages so I split it off into an own PR.

Fine for me.

**Comment by wamserma** [comment]
@SuperSandro2000 Thanks, updated.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108794` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.criticality-score</li>
<li>python38Packages.criticality-score</li>
<li>python39Packages.criticality-score</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108794` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.criticality-score</li>
<li>python38Packages.criticality-score</li>
<li>python39Packages.criticality-score</li>
</ul>
</details>

**Pull request merged**